current_spring2026/retrieval/retriever.py

```
@@ -0,0 +1,331 @@
+    dense_results: List[dict],
+    top_k: int = TOP_K_BM25,
+) -> List[dict]:
+    """Score sparse on dense candidates only — no extra DB call."""
+    if not query_sparse or not dense_results:
+        return []
+
+    scored = []
+    for row in dense_results:
+        token_ids    = row.get("sparse_token_ids") or []
+        weights      = row.get("sparse_weights")   or []
+        chunk_sparse = dict(zip([str(t) for t in token_ids], weights))
```

Ruff (0.15.10)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)